### ISSUE TITLE
Set multiple lenses successively

### ISSUE THREAD

**Issue by singularitti** [jw3126/Setfield.jl#74]
For example, I have the following code:
```julia
struct M
    c
end

struct N
    d
end

struct A
    a
    b
end

x = A(M(1), N(2))
lens1 = @lens _.a.c
lens2 = @lens _.b.d
```
If I want to set the two lenses successively, I need to write
```julia
set(set(x, lens1, 4), lens2, 5)
```
This code is a bit lacking readability. Could we provide a syntax that can set independent lenses successively? Thank you!

**Comment by tkf** [comment]
FYI I wrote `Kaleido.@batchlens` to do this. See: https://github.com/tkf/Kaleido.jl

**Comment by singularitti** [comment]
Wow, this is pretty cool!!! Thank you, I will take a look!

**Comment by jw3126** [comment]
Yeah Kaleido is pretty cool. I would love to have a batch `@set` someday.
